concent_api/core/message_handlers.py

```diff
@@ -400,7 +400,7 @@ def handle_send_force_get_task_result(client_message: message.concents.ForceGetT
         maximum_download_time
     )
 
-    if not client_message.report_computed_task.timestamp < get_current_utc_timestamp() <= force_get_task_result_deadline:
+    if not client_message.report_computed_task.timestamp <= get_current_utc_timestamp() <= force_get_task_result_deadline:
```

Comment:
If I understood: https://github.com/golemfactory/concent/issues/532 correctly, `client_message.report_computed_task.timestamp < ` part should be deleted 

@cameel ?

Reply:
Yes. Exactly. The task is about removing the check not about making the interval closed on both sides.

Reply:
In the way which it is described I have understood it in the way which @pawelkisielewicz did it, so maybe issue should be rewritten and add more info on technical level.

Reply:
> Is it possible that `client_message.report_computed_task.timestamp` would be greater then `get_current_utc_timestamp`?

It is and that's where this change comes from. Remember that the clocks of the provider, requestor and Concent might not be perfectly synchronized. Enforcing perfect synchronization is simply not realistic. That's for example why golem-messages, when checking if the timestamp is correct, has such a high tolerance: 10 minutes in the past and 5 minutes in the future: 

See
https://github.com/golemfactory/golem-messages/blob/v2.10.0/golem_messages/message/base.py#L38-L49
https://github.com/golemfactory/golem-messages/blob/v2.10.0/golem_messages/settings.py

``` python
MSG_TTL = datetime.timedelta(minutes=10)
FUTURE_TIME_TOLERANCE = datetime.timedelta(minutes=5)
```

So the timestamps received from another machine can be off by up to 10 minutes!

This means that for example when requestor is creating `ForceGetTaskResult` at 12:00, the `ReportComputedTask` may have a timestamp indicating 12:05.

There was a discussion between me, Przemek and Łukasz on Rocket and in #519 (please read them if you want to know more about why we're making this change) and this problem was raised. It has no easy fix if we want to treat the timestamps as exact. The "solution" is to simply never treat them as such on the client side. Golem client has some tolerances on message times. For example to be sure that it does not miss a deadline, it can't just send a message just a second before it. On another client or in Concent it may already be past the deadline at that point. To avoid this the client leaves enough room to be sure that even the maximum error coming from clocks being out of sync can't make it miss the deadline.

But in this case that's a bit inconvenient. We could take this approach but that would mean that sometimes the client would have to wait up to 5 mintues before sending FGTR. Fortunately there's a  workaround for this particular situation: we don't really have to check the timestamp on RCT. The only purpose of the lower limit is to make sure that the requestor does not try to start the use case before the provider has finished the computation. If the requestor has already received the RCT (he's including it in his FGTR) there's no reason to prevent him from starting the use case even if the timestamp on RCT is in the future.

Reply:
> In the way which it is described I have understood it in the way which @pawelkisielewicz did it, so maybe issue should be rewritten and add more info on technical level.

I have rewritten the description to include the information from the comment above.